Base/Logic/vtkSlicerApplicationLogic.h

```diff
@@ -229,10 +226,10 @@ class VTK_SLICER_BASE_LOGIC_EXPORT vtkSlicerApplicationLogic
   ~vtkSlicerApplicationLogic() override;
 
   /// Callback used by a std::thread to start a processing thread
-  static itk::ITK_THREAD_RETURN_TYPE ProcessingThreaderCallback( void * );
+  static void ProcessingThreaderCallback( void * );
```

Comment:
I am now wondering whether these functions can take `App *` instead of `void *`. That would allow us to avoid the type cast.

Reply:
Good even without this suggested change.

Reply:
Good point, I followed by with an additional commit further simplifying :rocket: 

In a nutshell, this is possible with C++11 and beyond thanks to:
- `std::thread` uses perfect forwarding (`std::forward`) to pass arguments efficiently
- `std::thread(F&& f, Args&&... args)` allows automatic argument binding